services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ProtectedPathSpec.groovy

```diff
@@ -0,0 +1,881 @@
+        northbound.portUp(islToBreakProtectedPath.dstSwitch.dpId, islToBreakProtectedPath.dstPort)
+        Wrappers.wait(WAIT_OFFSET + discoveryInterval) {
+            assert islUtils.getIslInfo(islToBreakProtectedPath).get().state == IslChangeType.DISCOVERED
+        }
+
+        then: "Path is not recalculated again"
+        pathHelper.convert(northbound.getFlowPath(flow.id).protectedPath) == newProtectedPath
+
+        and: "Cleanup: revert system to original state"
+        flowHelper.deleteFlow(flow.id)
+        database.resetCosts()
+    }
+
+    @Unroll
+    def "Unable to create protected path on #flowDescription flow on the same path"() {
+        given: "Two active neighboring switches with two not overlapping paths at least"
+        def (Switch srcSwitch, Switch dstSwitch) = getNeighboringSwitchPair(2)
+
+        and: "A flow without protected path"
+        def flow1 = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow1.allocateProtectedPath = false
+        flowHelper.addFlow(flow1)
+
+        and: "All alternative paths are unavailable (bring ports down on the source switch)"
+        List<PathNode> broughtDownPorts = []
+        database.getPaths(srcSwitch.dpId, dstSwitch.dpId)*.path.findAll {
+            it != pathHelper.convert(northbound.getFlowPath(flow1.id))
+        }.unique {
+            it.first()
+        }.each { path ->
+            def src = path.first()
+            broughtDownPorts.add(src)
+            northbound.portDown(src.switchId, src.portNo)
+        }
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northbound.getAllLinks().findAll {
+                it.state == IslChangeType.FAILED
+            }.size() == broughtDownPorts.size() * 2
+        }
+
+        when: "Try to create a new flow with protected path"
+        def flow2 = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow2.allocateProtectedPath = true
+        flow2.maximumBandwidth = bandwidth
+        flowHelper.addFlow(flow2)
+
+        then: "Human readable error is returned"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 404
+        //TODO investigate error message
+        exc.responseBodyAsString.to(MessageError).errorMessage ==
+                "Could not create flow: Not enough bandwidth found or path not found." +
+                " Couldn't find non overlapping protected path"
+
+        and: "Restore topology, delete flows and reset costs"
+        broughtDownPorts.every { northbound.portUp(it.switchId, it.portNo) }
+        flowHelper.deleteFlow(flow1.id)
+        Wrappers.wait(discoveryInterval + WAIT_OFFSET) {
+            northbound.getAllLinks().each { assert it.state != IslChangeType.FAILED }
+        }
+        database.resetCosts()
+
+        where:
+        flowDescription | bandwidth
+        "a metered"     | 1000
+        "an unmetered"  | 0
+    }
+
+    @Unroll
+    def "Unable to enable protected path on #flowDescription flow on the same path"() {
```

Comment:
Need to re-phrase this somehow.

Reply:
done
`"Unable to update #flowDescription flow to enable protected path on the same path`